### PR TITLE
Implements Ros-Control for Drivetrain

### PR DESCRIPTION
**Pull request by wmmc88**

### CODE REVIEW

File: uwrt_mars_rover_hw/src/uwrt_mars_rover_hw_drivetrain.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  joint_state_interface_.registerHandle(left_drive_joint);
+  hardware_interface::JointStateHandle right_drive_joint("RightDriveJoint", &position_[1], &velocity_[1], &effort_[1]);
+  joint_state_interface_.registerHandle(right_drive_joint);
+  registerInterface(&joint_state_interface_);
+
+  // connect and register the joint position_ interface
+  hardware_interface::JointHandle left_drive_position__handle(joint_state_interface_.getHandle("LeftDriveJoint"),
+                                                              &command_[0]);
+  joint_position_interface_.registerHandle(left_drive_position__handle);
+  hardware_interface::JointHandle right_drive_position__handle(joint_state_interface_.getHandle("RightDriveJoint"),
```
Comment (wraftus) [inline uwrt_mars_rover_hw/src/uwrt_mars_rover_hw_drivetrain.cpp]: two underscores in `left_drive_position__handle` :(

File: uwrt_mars_rover_hw/include/uwrt_mars_rover_hw/uwrt_mars_rover_hw_drivetrain_real.h [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#pragma once
+
+#pragma once
```
Comment (wraftus) [inline uwrt_mars_rover_hw/include/uwrt_mars_rover_hw/uwrt_mars_rover_hw_drivetrain_real.h]: should this be here twice?

File: uwrt_mars_rover_hw/src/uwrt_mars_rover_hw_drivetrain_real.cpp [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#include "uwrt_mars_rover_hw/uwrt_mars_rover_hw_drivetrain_real.h"
+
+namespace uwrt_mars_rover_hw {
+bool UWRTRoverHWDrivetrainReal::init(ros::NodeHandle& root_nh, ros::NodeHandle& robot_hw_nh) {
+  if (UWRTRoverHWDrivetrain::init(root_nh, robot_hw_nh)) {
```
Comment (wraftus) [inline uwrt_mars_rover_hw/src/uwrt_mars_rover_hw_drivetrain_real.cpp]: believe this should be `if (! ... ) {` ?

File: uwrt_mars_rover_hw/src/uwrt_mars_rover_hw_drivetrain.cpp [hunk 4]
```diff
@@ -0,0 +1,77 @@
+#include "uwrt_mars_rover_hw/uwrt_mars_rover_hw_drivetrain.h"
+
+namespace uwrt_mars_rover_hw {
+
+UWRTRoverHWDrivetrain::UWRTRoverHWDrivetrain() : UWRTRoverHWDrivetrain("UWRTRoverHWDrivetrain") {}
+
+bool UWRTRoverHWDrivetrain::init(ros::NodeHandle& root_nh, ros::NodeHandle& robot_hw_nh) {
+  bool param_fetched = robot_hw_nh.getParam("joints", joint_names_);
+  if (!param_fetched) {
```
Comment (wraftus) [inline uwrt_mars_rover_hw/src/uwrt_mars_rover_hw_drivetrain.cpp]: should return false if this fails?

File: uwrt_mars_rover_hw/include/uwrt_mars_rover_hw/uwrt_mars_rover_hw_drivetrain.h [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    double effort;
+  };
+
+  struct DrivetrainJointCommand {
+    enum class Type { NONE, POSITION, VELOCITY, EFFORT };
+    Type type;
+    double data;
+  };
+
+  bool init(ros::NodeHandle& root_nh, ros::NodeHandle& robot_hw_nh) override;
```
Comment (wraftus) [inline uwrt_mars_rover_hw/include/uwrt_mars_rover_hw/uwrt_mars_rover_hw_drivetrain.h]: should this be virtual too? i'm dumb tho so idk

Comment (wmmc88) [reply uwrt_mars_rover_hw/include/uwrt_mars_rover_hw/uwrt_mars_rover_hw_drivetrain.h]: nah, override already means theres a virtual function in the base class so while virtual can be added here, it is redundant. in fact, its a clang tidy waarning lol

File: uwrt_mars_rover_hw/src/uwrt_mars_rover_hw_drivetrain.cpp (hunk 4 above)
Comment (wmmc88) [reply uwrt_mars_rover_hw/src/uwrt_mars_rover_hw_drivetrain.cpp]: maybe, havent thought to much about it. it probably should

Comment (wmmc88) [reply uwrt_mars_rover_hw/src/uwrt_mars_rover_hw_drivetrain.cpp]: will fix in #39

**Pull request merged**